snuba/datasets/discover.py

```diff
@@ -66,6 +70,10 @@ def detect_dataset(query: Query, events_columns: ColumnSet, transactions_columns
 
 
 class DiscoverSource(RelationalSource):
```

Comment:
@lynnagara  do you remember any specific reason for nor making this reusable instead of discover specific ?
Also why does get_columns return the abstract datamodel instead of returning the columns of the selected table source? If there was no specific reason, we should change that so that a query processor always has access to the concrete schema that has been selected.

Reply:
Although DiscoverSource currently exists to facilitate switching between the two tables, this is likely to change a lot in future (for example earlier we talked about potentially merging results from multiple table sources). I'm not sure this kind of behaviour or whatever it ends up being is generically useful.

Yeah, get_columns was an oversight, it probably makes more sense if it reflects the columns of the selected table source.

Reply:
I think a general table switch will be useful. I can think about several cases where a query is dispatched to a separate table depending on the content. There is already a case that we have to build for outcomes.

Regarding get_columns, it *has* to reflect the columns of the table we are currently querying otherwise following processors would not be able to have a consistent view of the query they are tweaking. Unfortunately we do not yet have a good separation between the abstract dataset columnSet and the concrete clickhouse table one. These in the future should both be available to a query processors. But till then we should keep the Query object consistent.